ui/src/components/App/App.jsx

```diff
@@ -169,6 +168,18 @@ export default function App() {
     
   }
 
+
+  // Toggles between displaying an icon that links to the home page (displays if you 
+  // are in the user profile) and an icon that links to the user profile (displays if
+  // you are on the homepage). This function is sent in as a prop to the NavBar component
+  function toggleViewProfile() {
+    if (viewProfile) {
+      setViewProfile(false);
+    }
+    else {
+      setViewProfile(true);
+    }
```

Comment:
for react state toggling, you want to do this.
setViewProfile(prevView => !prevView)